From 52aa356ba91e47df68fe1d9dac3f4f0605cda71f Mon Sep 17 00:00:00 2001
From: Abhishek Srivastava <cabsriv@codeaurora.org>
Date: Fri, 17 Aug 2018 12:22:06 +0530
Subject: [PATCH 3/4] Wi-Fi : Abort setWifiEnabled if WifiController not ready

While executing setWifiEnabled, it send commands to WifiController.
If WifiController is not started by this time, StateMachine reports
fatal which creates device reboot.
Thus, check if WiFiController is started before invoking any
command to the same. Indicate a failure in such an instance

Change-Id: I174b8787cbc61d9e62a4cfefc9885f73f3f14775
CRs-Fixed: 2274025
---
 service/java/com/android/server/wifi/WifiServiceImpl.java | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/service/java/com/android/server/wifi/WifiServiceImpl.java b/service/java/com/android/server/wifi/WifiServiceImpl.java
index c38aa1c66..1c8f1b2a5 100644
--- a/service/java/com/android/server/wifi/WifiServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiServiceImpl.java
@@ -168,6 +168,8 @@ public class WifiServiceImpl extends IWifiManager.Stub {
     private WifiTrafficPoller mTrafficPoller;
     /* Tracks the persisted states for wi-fi & airplane mode */
     final WifiSettingsStore mSettingsStore;
+
+    private boolean mIsControllerStarted = false;
     /* Logs connection events and some general router and scan stats */
     private final WifiMetrics mWifiMetrics;
 
@@ -613,6 +615,7 @@ public class WifiServiceImpl extends IWifiManager.Stub {
             Log.wtf(TAG, "Failed to initialize WifiStateMachine");
         }
         mWifiController.start();
+        mIsControllerStarted = true;
 
         // If we are already disabled (could be due to airplane mode), avoid changing persist
         // state here
@@ -904,6 +907,11 @@ public class WifiServiceImpl extends IWifiManager.Stub {
             }
         }
 
+        if (!mIsControllerStarted) {
+            Slog.e(TAG,"WifiController is not yet started, abort setWifiEnabled");
+            return false;
+        }
+
         mWifiController.sendMessage(CMD_WIFI_TOGGLED);
         return true;
     }
-- 
2.17.1

